File: pascalTriangleII/pascalTriII.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<List<Integer>> nums = new ArrayList<List<Integer>>();
        for(int i=0;i<rowIndex+1;i++){
            List<Integer> num = new ArrayList<Integer>();
            for(int j=0;j<i+1;j++){
                if(j==0 || j == i) num.add(1);
                else{
                   num.add(nums.get(i-1).get(j-1) + nums.get(i-1).get(j)); 
                } 
            }
            nums.add(num);
        }
        return nums.get(rowIndex);
    }
}    

File: pascalTriangleII/pascalTriII.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        if(rowIndex<0):rowIndex = 0
        nums = [[0 for j in range(i+1)] for i in range(rowIndex+1)]
        for i in range(0,rowIndex+1):
            for j in range(0,i+1):
                if j==0 or j==i: nums[i][j] = 1
                else: nums[i][j] = nums[i-1][j-1]+nums[i-1][j]
            i+=1
        return nums[rowIndex][:]
            